src/PlusProducts.php

```diff
@@ -51,7 +51,7 @@ public static function woocommercePlusProductsSettings(array $settings): array {
   public static function displayPlusProductsNotice() {
     $plusCategory = get_term_by('id', get_option('_' . Plugin::L10N . '_plus_products_category'), 'product_cat');
 
-    if (is_wp_error($plusCategory) || empty($plusCategory)) {
+    if (is_wp_error($plusCategory) || empty($plusCategory) || WC()->cart->is_empty()) {
       return;
     }
```

Comment:
Why not check first if cart empty and do an early return? Chances are that the category term query is not needed in some cases.
```
if (WC()->cart->is_empty()) {
  return;
}
$plusCategory = get_term_by('id', get_option('_' . Plugin::L10N . '_plus_products_category'), 'product_cat');
if (empty($plusCategory) || is_wp_error($plusCategory)) {
  return;
}

Reply:
Yeah, sure. Good catch, thanks. 👍